Controller/UserController.php

```diff
@@ -255,6 +257,11 @@ public function apiKeyAction($username, LoggerInterface $logger)
     {
         $user = $this->getUserService()->getUser($username, false);
 
+        $roles = $user->getRoles();
+        if (!in_array('ROLE_API', $roles)) {
+            throw new \RuntimeException('The user "' . $username . '" does not have the permission to use API functionalities.');
+        }
```

Comment:
Could you:
- Add an entry in the logger entry: 
            $this->logger->error('log.user.cannot_request_api_key', [
                'user' => $username,
            ]);
- Use the sprintf function to generate the exception message

Reply:
@theus77 OK done, I also included the initiator in the log.